cores/esp8266/core_esp8266_i2s.c

```diff
@@ -208,6 +333,33 @@ bool ICACHE_FLASH_ATTR i2s_write_lr(int16_t left, int16_t right){
   return i2s_write_sample(sample);
 }
 
+bool ICACHE_FLASH_ATTR i2s_read_sample(uint32_t *left, uint32_t *right, bool blocking) {
+  if (rx->curr_slc_buf_pos==SLC_BUF_LEN || rx->curr_slc_buf==NULL) {
+    if (rx->slc_queue_len == 0) {
+      if (!blocking) return false;
+      while (1) {
+        if (rx->slc_queue_len > 0){
+          break;
+        } else {
+          ets_wdt_disable();
+          ets_wdt_enable();
+        }
+      }
+    }
+    ETS_SLC_INTR_DISABLE();
+    rx->curr_slc_buf = (uint32_t *)i2s_slc_queue_next_item(rx);
+    ETS_SLC_INTR_ENABLE();
+    rx->curr_slc_buf_pos=0;
+  }
+
+//  *left  = rx->slc_items[0].buf_ptr[rx->curr_slc_buf_pos++];
```

Comment:
Can this commented code be removed, or is it some reference?

Reply:
My debugging, the DMA buffer handoff is still funky in the code and on the Rx side the receiver can be overwriting the block you're reading from.  It's to be fixed before real submission.